packages/website/docs/sources.md

```diff
@@ -180,15 +167,50 @@ const autocomplete = createAutocomplete({
 });
 ```
 
-You can notice that we can **get the `query` from the `getSources` function to conditionally return sources**.
+The [`getSources`](#getsources) function provides access to the current `query`, which you can use to return sources conditionally. You can use this pattern to display recent searches when the query is empty and search results when the user types.
 
-This pattern can be extended to display recent searches on empty query, and search results when a query is typed. You can compute dynamic sources based on the complete autocomplete state, not only the query.
+Note that you have access to the [complete autocomplete state](state), not only the query. It lets you compute sources based on many different aspects.
```

Comment:
- "complete autocomplete" sound weird, don't you think?
- "aspects" is vague

```suggestion
Note that you have access to the [whole autocomplete state](state), not only the query. It lets you compute sources based on many different criteria.
```

Reply:
Right, I've switched with "full", and clarified the sentence with examples and a link to `state#state` in 9441f28.